backend/src/routes/upload.ts

```diff
@@ -86,7 +86,18 @@ export class UploadRoutes {
       }
 
       const params = { site: site, measurementDate: body.measurementDate, filename: filename };
-      const payload = instrumentUpload ? { ...params, instrument: body.instrument } : { ...params, model: body.model };
+      let instrumentParams;
+      let sortedTags;
+      if (instrumentUpload) {
+        const allowedTags = new Set((dataSource as Instrument).allowedTags);
+        const metadataTags = new Set(body.tags as Array<string>);
+        sortedTags = Array.from(new Set(Array.from(metadataTags).filter((x) => allowedTags.has(x)))).sort();
```

Comment:
Tämän pitäisi tehdä sama asia lyhyemmin, koska duplikaatit on jo poistettu:

```suggestion
        sortedTags = Array.from(metadataTags).filter((x) => allowedTags.has(x)).sort();
```